ilastik/applets/neuralNetwork/bioimageiodl.py

```diff
@@ -0,0 +1,91 @@
+import pathlib
+from io import BytesIO
+from zipfile import ZIP_DEFLATED
+
+from bioimageio.core import load_raw_resource_description
+from bioimageio.core.resource_io.io_ import make_zip
+from bioimageio.spec import get_resource_package_content
+from bioimageio.spec.shared import resolve_source, raw_nodes
+from PyQt5.QtCore import QThread, pyqtSignal
+from tqdm.auto import tqdm as std_tqdm
+
+from functools import partial
+
+from asyncio import CancelledError
+
+import logging
+
+logger = logging.getLogger(__file__)
+
+BIOIMAGEIO_WEIGHTS_PRIORITY = ["torchscript", "pytorch_state_dict", "tensorflow_saved_model_bundle"]
+
+
+class TqdmExt(std_tqdm):
+    def __init__(self, *args, callback=None, cancellation_token=None, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._cb = callback
+        self._cancellation_token = cancellation_token
+
+    def update(self, n=1):
+        if self._cancellation_token:
+            if self._cancellation_token.cancelled:
+                raise CancelledError()
+
+        displayed = super(TqdmExt, self).update(n)
+        if displayed:
+            if self._cb:
+                self._cb(**self.format_dict)
+        return displayed
+
+
+class BioImageDownloader(QThread):
```

Comment:
since progress is communicated in the gui, signals are appropriate -> `QThread`. This function will only be used in gui mode - for headless mode, bioimage libraries are sufficient by themselves.